### PR TITLE
Add remove unused variable #23

### PR DESCRIPTION
**Pull request by julianosaless**
Remove unused variables

### CODE REVIEW

File: src/CSharp/CodeCracker/Usage/RemoveUnusedVariablesAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            if (methodOrConstructor?.Body?.Statements.FirstOrDefault() == null || methodOrConstructor?.Body?.Statements.LastOrDefault() == null) return;
+
+            var dataFlowAnalysis = semanticModel.AnalyzeDataFlow(methodOrConstructor?.Body?.Statements.FirstOrDefault(), methodOrConstructor?.Body?.Statements.LastOrDefault());
+
+            foreach (var variable in variables)
+            {
+                if (!IsUsedVariable(variable, semanticModel, dataFlowAnalysis))
+                {
+                    ReportDiagnostic(context, variable);
+                    return;
```
Comment (viniciushana) [inline src/CSharp/CodeCracker/Usage/RemoveUnusedVariablesAnalyzer.cs]: Wouldn't it just report one diagnostic and exit the method altogether? Perhaps it's better to report all diagnostics without a return statement, letting the foreach loop run for all variables in this context.

Comment (julianosaless) [inline src/CSharp/CodeCracker/Usage/RemoveUnusedVariablesAnalyzer.cs]: @viniciushana  really makes sense, i´ll remove the return :)

**Comment by viniciushana** [comment]
Looks good!

**Comment by giggio** [comment]
I don't like that we are running this analyzer again. The C# compiler already checks for this with `CS0168`: https://msdn.microsoft.com/en-us/library/tf85t6e4%28v=vs.90%29.aspx
You can delete the analyzer and only use the code fix, @julianosaless. Can you fix that?

**Comment by giggio** [comment]
@viniciushana You merged an issue @carloscds was already working on.

**Comment by carloscds** [comment]
@giggio I was looking, but @viniciushana commited.

**Comment by viniciushana** [comment]
Sorry about that, didn't pay attention.

**Comment by julianosaless** [comment]
@giggiomy analyzer is better  kkkkk,  i´ll remove it :) sorry

**Comment by carloscds** [comment]
@julianosaless did you keep fixer ?

**Comment by julianosaless** [comment]
@carloscds i´ll remove only  analyzer.

**Comment by julianosaless** [comment]
Não gostei muito da maneira que a  C# compiler desenvolveu, 
ele possue  2 analyzer, quando uma variável tem valor e quando uma variável não tem, o analyzer que eu desenvolvi engloba as duas possibilidades abaixo: 

CS1688 : The variable 'var' is assigned but its value is never used

public class clz
{
   public static void Main()
   {
      int j = 0;   // CS0168, uncomment the following line
      // j++;
      clx a;       // CS0168, try the following line instead
      // clx a = new clx();
   }
}


CS0219 :  The variable 'variable' is assigned but its value is never used

public class MyClass
{
   public static void Main()
   {
      int a = 0;   // CS0219
   }
}

Porém  no meu Fix eu vou atribuir os dois analyzer para ele: ImmutableArray.Create(new[] { "CS0168", "CS0219" });

**Comment by giggio** [comment]
Closes #23.

**Pull request merged**